src/util/read-pjson.ts

```diff
@@ -0,0 +1,64 @@
+import {cosmiconfig} from 'cosmiconfig'
+import {join} from 'node:path'
+
+import {PJSON} from '../interfaces'
+import {makeDebug} from '../logger'
+import {readJson} from './fs'
+
+const debug = makeDebug('read-pjson')
+
+/**
+ * Read the package.json file from a given path and merge with the config found by cosmiconfig.
+ *
+ * We can assume that the package.json file exists because the plugin root has already been loaded at this point.
+ */
+export async function readPjson(path: string): Promise<PJSON.Plugin> {
+  const pjsonPath = join(path, 'package.json')
+  if (process.env.OCLIF_DISABLE_RC) {
+    debug('OCLIF_DISABLE_RC is set, skipping rc search')
+    return readJson<PJSON.Plugin>(pjsonPath)
+  }
+
+  const pjson = await readJson<PJSON.Plugin>(pjsonPath)
+
+  // don't bother with cosmiconfig if the plugin's package.json already has an oclif config
+  if (pjson.oclif) {
+    debug(`found oclif config in ${pjsonPath}`)
+    return pjson
+  }
+
+  debug(`searching for oclif config in ${path}`)
+  const explorer = cosmiconfig('oclif', {
+    /**
+     * Remove the following from the defaults:
+     * - package.json
+     * - any files under .config/
+     */
+    searchPlaces: [
+      '.oclifrc',
+      '.oclifrc.json',
+      '.oclifrc.yaml',
+      '.oclifrc.yml',
+      '.oclifrc.js',
+      '.oclifrc.ts',
+      '.oclifrc.mjs',
+      '.oclifrc.cjs',
+      'oclif.config.js',
+      'oclif.config.ts',
+      'oclif.config.mjs',
+      'oclif.config.cjs',
```

Comment:
Is there a good use-case to support so many filetypes here? I feel like just supporting `.oclifrc.json` would be plenty. Then we could get rid off the `cosmicconfig` package.

Reply:
`js` and `ts` files offer several benefits:

- comments
- dynamically set props
- use types